extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergBaseLayout.java

```diff
@@ -130,6 +132,14 @@ public IcebergBaseLayout(
 
     abstract IcebergTableLocationKey keyFromDataFile(DataFile df, URI fileUri);
 
+    private URI dataFileUri(DataFile df) {
```

Comment:
You can follow the same `@NotNull final` convention as done in other methods of this file.